### DISCUSSION TITLE
How to clean multipart dirty data

### DISCUSSION THREAD

**Discussion by alixiaodi** [minio/minio#16228] [Q&A] [answered]
I have used multipart to upload some data successfully. At this time, the client cancels the upload. The multipart data that has been uploaded successfully is unavailable. How can I clean it up?

**Comment by harshavardhana** [comment, marked as answer]
No need to do that, it is automatically cleared every 24hours you can change that to be faster

```
stale_uploads_expiry            (duration)  set to expire stale multipart uploads older than this values (default: '24h')
stale_uploads_cleanup_interval  (duration)  set to change intervals when stale multipart uploads are expired (default: '6h')
```

**Reply by alixiaodi** [reply]
I see Thank you

**Reply by alixiaodi** [reply]
For example, I now have a 100TB file uploaded, but it cannot be uploaded for 24 hours. At this time, the purge policy will be triggered. Can I only solve this problem by modifying the purge policy from time to time

**Reply by harshavardhana** [reply]
There is no 100TiB file allowed in AWS S3 API @alixiaodi you are asking for hypothetical scenarios.

**Reply by alixiaodi** [reply]
Yes, I want to express that my actual upload time has exceeded 24 hours, and the removal strategy should not delete it, because I am still uploading, do I have to configure it to take longer? Is that right, or is there another solution

**Reply by klauspost** [reply]
If you see the need you can extend the deadline as described above.

**Reply by alixiaodi** [reply]
OK, I see. Thank you

**Comment by harshavardhana** [comment]
> The multipart data that has been uploaded successfully is unavailable. How can I clean it up?

You should have called AbortMultipartUpload() with the uploadID if you want an immediate cleanup.
